cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_ctrl_packages.go

```diff
@@ -343,3 +346,85 @@ func (s *Server) GetInstalledPackageSummaries(ctx context.Context, request *core
 	}
 	return response, nil
 }
+
+// GetInstalledPackageDetail returns the package metadata managed by the 'kapp_controller' plugin
+func (s *Server) GetInstalledPackageDetail(ctx context.Context, request *corev1.GetInstalledPackageDetailRequest) (*corev1.GetInstalledPackageDetailResponse, error) {
+	log.Infof("+kapp-controller GetInstalledPackageDetail")
+
+	// Retrieve the proper parameters from the request
+	cluster := request.GetInstalledPackageRef().GetContext().GetCluster()
+	namespace := request.GetInstalledPackageRef().GetContext().GetNamespace()
+	installedPackageRefId := request.GetInstalledPackageRef().GetIdentifier()
+
+	if cluster == "" {
+		cluster = s.globalPackagingCluster
+	}
+
+	typedClient, _, err := s.GetClients(ctx, cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	// fetch the package install
+	pkgInstall, err := s.getPkgInstall(ctx, cluster, namespace, installedPackageRefId)
+	if err != nil {
+		return nil, errorByStatus("get", "PackageInstall", installedPackageRefId, err)
+	}
+
+	// fetch the resulting deployed app after the installation
+	app, err := s.getApp(ctx, cluster, namespace, installedPackageRefId)
+	if err != nil {
+		return nil, errorByStatus("get", "App", installedPackageRefId, err)
+	}
+
+	// retrieve the package metadata associated with this installed package
+	pkgName := pkgInstall.Spec.PackageRef.RefName
+	pkgMetadata, err := s.getPkgMetadata(ctx, cluster, namespace, pkgName)
+	if err != nil {
+		return nil, errorByStatus("get", "PackageMetadata", pkgName, err)
+	}
+
+	// Use the field selector to return only Package CRs that match on the spec.refName.
+	fieldSelector := fmt.Sprintf("spec.refName=%s", pkgMetadata.Name)
+	pkgs, err := s.getPkgsWithFieldSelector(ctx, cluster, namespace, fieldSelector)
+	if err != nil {
+		return nil, errorByStatus("get", "Package", "", err)
+	}
+	pkgVersionsMap, err := getPkgVersionsMap(pkgs)
+	if err != nil {
+		return nil, err
+	}
+
+	// get the values applies i) get the secret name where it is stored; 2) get the values from the secret
+	valuesApplied := ""
+	// retrieve every value and build a single string containing all of them
+	for _, pkgInstallValue := range pkgInstall.Spec.Values {
+		secretRefName := pkgInstallValue.SecretRef.Name
+		// if there is a secret containing the applied values of this installed package, get the them
+		if secretRefName != "" {
+			values, err := typedClient.CoreV1().Secrets(namespace).Get(ctx, secretRefName, metav1.GetOptions{})
+			if err != nil {
+				if errors.IsNotFound(err) {
+					log.Warningf("The referenced secret does not exist: %s", errorByStatus("get", "Secret", secretRefName, err).Error())
```

Comment:
Why don't we return an error here? (where the package install references a secret that doesn't exist).

Reply:
It's for the same reason as in previous comments, I've prioritized "business continuity" here, so in case a user manually deletes the secrets, I think the app should be still manageable by Kubeapps, at least for deleting it.
Happy to seek more alternatives, though.